src/CentrifugeRouter.sol

```diff
@@ -45,11 +50,13 @@ contract CentrifugeRouter is Auth, Multicall, ICentrifugeRouter {
     }
 
     // --- Deposit ---
-    function requestDeposit(address vault, uint256 amount) external payable {
-        require(msg.value > 0, "CentrifugeRouter/not-enough-gas-funds");
-        // Due to the multicall nature of this contract, there are issues calling `transfer` multiple times.
-        // Investigate further.
-        payable(gateway).call{value: msg.value}("");
+    function requestDeposit(address vault, uint256 amount, uint256 topUpAmount) external payable {
+        // Assumption: This contract won't hold any ETH
+        require(topUpAmount <= msg.value, "CentrifugeRouter/must-pay-for-tx-costs");
+        // In a situation where we have multiple calls ( as in multicall ) the msg.value remains the same
+        // but with each call the balance will get reduced by `topUp`
+        require(topUpAmount <= address(this).balance, "CentrifugeRouter/insufficient-funds-to-topup");
```

Comment:
I don't understand this check. Even if you call `requestDeposit` multiple times within 1 tx, you would still need gas for multiple tx to send multiple requests to cfg chain

Reply:
Check my comment for the previous line.